### PR TITLE
WFLY-7887 Add missing cache configuration dependencies for custom cache regions.

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-7887

### CODE REVIEW

File: jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java [hunk 1]
```diff
@@ -63,28 +78,50 @@ public static void addSecondLevelCacheDependencies(Properties mutableProperties,
...
+
+            // Collect distinct cache configurations for custom regions
+            Set<String> customRegionConfigs = mutableProperties.stringPropertyNames().stream()
+                    .filter(name -> name.startsWith(customRegionPrefix) && name.endsWith(customRegionSuffix))
+                    .map(name -> mutableProperties.getProperty(name))
+                    .filter(config -> !standardRegionConfigs.contains(config))
+                    .collect(Collectors.toSet());
+
+            if (!customRegionConfigs.isEmpty()) {
+                cacheSettings.setProperty(CUSTOM, String.join(" ", customRegionConfigs));
```
Comment (scottmarlow) [inline jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java]: Hi Paul, does the [ORM InfinispanRegionFactory class](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java) read the "custom" property?  Or something else?

Thanks,
Scott

Comment (pferraro) [inline jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java]: This property is only read by org.jboss.as.jpa.processor.secondLevelCache.InfinispanCacheDeploymentListener in order to establish the requisite service dependencies.

Comment (scottmarlow) [inline jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java]: Ahh, thanks,  I get it now.  The change looks good, thanks for making it!

**Comment by kabir** [comment]
Retest this please

**Pull request merged**